docs/modules/ROOT/pages/integrations/database.adoc

```diff
@@ -0,0 +1,69 @@
+= Database Integrations
+
+== Overview
+
+KubeArchive needs a database to store the resources to make them retrievable.
+KubeArchive currently supports integration with PostgreSQL.
+MariaDB integration is in progress.
+
+[NOTE]
+====
+
+KubeArchive integration test suite runs with PostgreSQL 16.6
+
+====
+
+== Configuration and Customization
+
+KubeArchives offers a [filename]`kubearchive.sql` file with the DDL to create
+the schema as part of the installation.
+
+[IMPORTANT]
+====
+The provided schema includes indexes.
+
+KubeArchive users should review and customize the indexes
+if needed based on the expected queries and the amount and
+the distribution of the expected archived data.
+====
+
+Specific instructions for the configuration of the PostgreSQL Database are available in
+xref:/getting-started/instllation.adoc#_install_and_configure_kubearchive[Install and Configure KubeArchive].
+
+== Support new databases
+
+KubeArchive database integration is designed to include new databases.
+
+To add a new database integration the following interfaces need to be implemented:
+
+* link:https://github.com/kubearchive/kubearchive/blob/main/pkg/database/facade/creator.go[DBCreator]
+* link:https://github.com/kubearchive/kubearchive/blob/main/pkg/database/facade/deleter.go[DBDeleter]
+* link:https://github.com/kubearchive/kubearchive/blob/main/pkg/database/facade/filter.go[DBFilter]
+* link:https://github.com/kubearchive/kubearchive/blob/main/pkg/database/facade/inserter.go[DBInserter]
+* link:https://github.com/kubearchive/kubearchive/blob/main/pkg/database/facade/selector.go[DBSelector]
+* link:https://github.com/kubearchive/kubearchive/blob/main/pkg/database/facade/sorter.go[DBSorter]
+
+[NOTE]
+====
+Except for the `DBCreator` functions, all the functions in the interfaces receive/output objects of the
+link:https://pkg.go.dev/github.com/huandu/go-sqlbuilder[sqlbuilder Go library].
+Check out the docs to see how to work with them.
+====
+
+[TIP]
+====
+Some of the interfaces like `DBDeleter` include a default implementation that can be used called `DBDeleterImpl`.
+
+Other interfaces like `DBFilter` include a partial implementation called `PartialDBFilterImpl` with the
+implementations of the filters that are likely to be reused.
+
+Check if those default implementations fit your database integration needs before implementing them.
+====
+
+If more customization is needed, override the implementation of the functions declared in
+link:https://github.com/kubearchive/kubearchive/blob/main/pkg/database/database.go#L29[DBInterface].
+
+The PostgreSQL implementation is available in
+link:https://github.com/kubearchive/kubearchive/blob/main/pkg/database/postgresql.go[here].
```

Comment:
If you liked the idea of having a section per database, this should go into the PostgreSQL section

Reply:
done!